compiler/rustc_middle/src/ty/util.rs

```diff
@@ -786,6 +786,13 @@ impl<'tcx> TyCtxt<'tcx> {
             || self.extern_crate(key.as_def_id()).is_some_and(|e| e.is_direct())
     }
 
+    /// Whether the item has a host effect param. This is different from `TyCtxt::is_const`,
+    /// because the item must also be "maybe const", and the crate where the item is
+    /// defined must also have the effects feature enabled.
+    pub fn has_host_param(self, def_id: impl Into<DefId>) -> bool {
```

Comment:
you could theoretically relax this to `impl IntoQueryParam<DefId>` to avoid the `.into()` call in the body

Reply:
I don't know if `IntoQueryParam` is actually an accessible trait from here. I was always under the impression that it's private to the query system implementation -- otherwise I'd use it far more aggressively.